src/Compilers/CSharp/Test/Semantic/Semantics/ExtensionTypeTests.cs

```diff
@@ -8158,4 +8232,1151 @@ public void ObsoleteOnType()
         // PROTOTYPE revisit once attributes are allowed on extension types.
         // The Obsolete poison attribute should not be emitted when the user marked the type as obsolete.
     }
+
+    [ConditionalFact(typeof(CoreClrOnly))]
+    public void MemberLookup()
+    {
+        var src = """
+E.M();
+_ = E.Property;
+
+public explicit extension E for object
+{
+    public static void M()
+    {
+        System.Console.Write("Method ");
+    }
+
+    public static int Property
+    {
+        get
+        {
+            System.Console.Write("Property");
+            return 0;
+        }
+    }
+}
+""";
+
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+        comp.VerifyDiagnostics();
+        CompileAndVerify(comp, expectedOutput: "Method Property",
+            symbolValidator: validate, sourceSymbolValidator: validate, verify: Verification.FailsPEVerify);
+
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+        var invocation = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().First();
+        Assert.Equal("E.M()", invocation.ToString());
+        Assert.Equal("void E.M()", model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+
+        var property = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().Skip(1).First();
+        Assert.Equal("E.Property", property.ToString());
+        Assert.Equal("System.Int32 E.Property { get; }", model.GetSymbolInfo(property).Symbol.ToTestDisplayString());
+
+        void validate(ModuleSymbol module)
+        {
+            bool inSource = module is SourceModuleSymbol;
+            var e = module.GlobalNamespace.GetTypeMember("E");
+            Assert.Empty(e.BaseExtensionsNoUseSiteDiagnostics);
+            Assert.Empty(e.AllBaseExtensionsNoUseSiteDiagnostics);
+        }
+    }
+
+    [ConditionalFact(typeof(CoreClrOnly), AlwaysSkip = "fails in emit stage in used assemblies leg")]
+    public void MemberLookup_Instance()
+    {
+        var src = """
+void M(E e)
+{
+    e.M();
+    _ = e.Property;
```

Comment:
>_ = e.Property; [](http://example.com/codeflow?start=4&length=15)

Are we testing an attempt to find a member of the underlying type? That would be especially interesting when the underlying type is not ```System.Object```.

Reply:
Yes, those are planned, but not in this PR. I only included one test to hint at this area (see PROTOTYPE marker in `MemberLookup_MethodsFromObject`).